File: greedy/1558_minimum_numbers_of_function_calls_to_make_target_array/1558_minimum_numbers_of_function_calls_to_make_target_array.py
from typing import List

class Solution:
    def minOperations(self, nums: List[int]) -> int:
        count = 0
        while any(nums):
            for i in range(len(nums)):
                if nums[i] % 2:
                    nums[i] -= 1
                    count += 1
            if any(nums):
                for i in range(len(nums)):
                    nums[i] //= 2
                count += 1
        return count


nums  = [1,5]
res = Solution().minOperations(nums)
print(res)

File: greedy/1558_minimum_numbers_of_function_calls_to_make_target_array/1558_minimum_numbers_of_function_calls_to_make_target_array.go
package main

import "fmt"

func minOperations(nums []int) int {
	tot := 0
	maxCounter := 0
	for _, v := range nums {
		counter := 0
		for v > 1 {
			if v%2 == 1 {
				v--
				tot++
			}
			v /= 2
			counter++
		}

		if v == 1 {
			tot++
		}

		if maxCounter < counter {
			maxCounter = counter
		}
	}

	return tot + maxCounter
}

func main() {
	nums := []int{1, 5}
	res := minOperations(nums)
	fmt.Println(res)
}
